src/systems/filecoin_mining/storage_mining/storage_miner_actor.go

```diff
@@ -574,19 +574,26 @@ func (a *StorageMinerActorCode_I) DeclareFaults(rt Runtime, faultSet sector.Comp
 }
 
 func (st *StorageMinerActorState_I) _isSealVerificationCorrect(rt Runtime, onChainInfo sector.OnChainSealVerifyInfo) bool {
-	// TODO: verify seal @nicola
-	// TODO: st.verifySeal(sectorID SectorID, comm sector.OnChainSealVerifyInfo, proof SealProof)
-
-	// verifySeal will also generate CommD on the fly from CommP and PieceSize
+	sectorSize := st.Info().SectorSize()
+	dealIDs := onChainInfo.DealIDs()
+	params := make([]actor.MethodParam, 1+len(dealIDs))
+	// TODO: serialize method param as {sectorSize,  DealIDs...}.
+
+	receipt := rt.Send(&msg.InvocInput_I{
+		To_:     addr.StorageMarketActorAddr,
+		Method_: actor.MethodGetUnsealedCIDForDealIDs,
+		Params_: params,
+		Value_:  rt.ValueSupplied(), // TODO: figure out where tx fee should come from
+	})
 
-	var pieceInfos []sector.PieceInfo // = make([]sector.PieceInfo, 0)
+	if receipt.ExitCode() == exitcode.InvalidSectorPacking {
```

Comment:
By default, a failure exit code in a `Send` call will cascade, causing the current method to also fail immediately with a `MethodSubcallError`. To prevent this and check the error code manually, one must call `SendAllowingErrors` instead of `Send`.